### PR TITLE
MODE-2549 Adds PDF sequencer

### PR DESCRIPTION
**Pull request by RobSis**
Initial implementation of PDF sequencer based on Apache PDFBox.

### CODE REVIEW

File: sequencers/modeshape-sequencer-pdf/src/main/java/org/modeshape/sequencer/pdf/PdfBasicMetadata.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                keywords = metadata.getKeywords();
+                modificationDate = metadata.getModificationDate();
+                producer = metadata.getProducer();
+                subject = metadata.getSubject();
+                title = metadata.getTitle();
+
+                document.close();
+            }
+            return true;
+        } catch (IOException | IllegalArgumentException e) {
```
Comment (hchiorean) [inline sequencers/modeshape-sequencer-pdf/src/main/java/org/modeshape/sequencer/pdf/PdfBasicMetadata.java]: these should either 
a) be logged here and therefore the `logger` object from the parent sequencer passed in and used here
b) left to bubble up and caught and logged by the sequencer

in either case we want to be aware that these exceptions occurred.

File: sequencers/modeshape-sequencer-pdf/src/main/java/org/modeshape/sequencer/pdf/PdfBasicMetadata.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * Set the input stream to the argument stream (or file).
+     *
+     * @param inputStream the input stream to read from
+     */
+    public void setInput( InputStream inputStream ) {
```
Comment (hchiorean) [inline sequencers/modeshape-sequencer-pdf/src/main/java/org/modeshape/sequencer/pdf/PdfBasicMetadata.java]: this seems to be always required, so it should be converted to a constructor

File: sequencers/modeshape-sequencer-pdf/src/main/resources/org/modeshape/sequencer/pdf/pdf.cnd [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  - pdf:creationDate (date)
+  - pdf:creator (string)
+  - pdf:keywords (string)
+  - pdf:modificationDate (date)
+  - pdf:producer (string)
+  - pdf:subject (string)
+  - pdf:title (string)
+  + pdf:xmp (pdf:xmp) = pdf:xmp
+  + pdf:attachment = pdf:xmp
+  + pdf:page = pdf:xmp
```
Comment (maschmid) [inline sequencers/modeshape-sequencer-pdf/src/main/resources/org/modeshape/sequencer/pdf/pdf.cnd]: Note, the current code doesn't seem to be creating the pdf:page and pdf:attachment nodes.

**Comment by hchiorean** [comment]
any status update on this ? ModeShape 4.6.Final will be released next week and most likely will be the last 4.x release which will include new features.

Everything else will get moved to ModeShape 5.

**Comment by RobSis** [comment]
Thanks for the input. 
I'm fine with moving it to v5, since I won't have too much time to work on it in next few weeks.

**Comment by hchiorean** [comment]
@RobSis: are there any plans of updating this in the near future ? We're evaluating the state of our open enhancement requests so we need to know whether to consider or close this

**Comment by RobSis** [comment]
Hello. Yes! 
Didn't have a lot of time, but I'm working on it and it should be ready in next week. 

Thank you for notice!

**Pull request closed without merging**